libraries/chef_ingredient_resource.rb

```diff
@@ -21,22 +21,22 @@ class ChefIngredient < Chef::Resource::LWRPBase
 
       actions :install, :uninstall, :remove, :reconfigure, :upgrade
       default_action :install
-      state_attrs :installed
 
       attribute :product_name, kind_of: String, name_attribute: true
-      attribute :installed, kind_of: [TrueClass, FalseClass, NilClass], default: false
-      attribute :reconfigure, kind_of: [TrueClass, FalseClass], default: false
       attribute :config, kind_of: String, default: nil
 
+      # Attributes for determining what version to install from which channel
+      attribute :version, kind_of: [String, Symbol], default: :latest
+      attribute :channel, kind_of: Symbol, default: :stable, equal_to: [:current, :stable]
```

Comment:
Yeah, I'm kind of hugely :-1: on that kind of sugar based on the abuse of attributes and other areas.  There was some `#to_sym` stuff in the old `provides` which I turned into a deprecation warning in Chef 12.0, it just generates more testing and ambiguity and we should just define which the interface takes and force the user to use the one we pick.